### PR TITLE
fix(sdk): unable to add multiple bucket notifications

### PR DESCRIPTION
**Pull request by tsuf239**
fixes: https://github.com/winglang/wing/issues/2308

Fixes a bug in which the addition of a few event notifications to the same bucket results in the creation of only one event notification when deploying to AWS (even if there are a few event notifications appearing in Terraform)

image

apparently, it's a known issue while using Terraform- 
 and is fixed by keeping only s3BucketNotification per bucket, and adding an id to each one of the s3BucketNotification's topics.

image

(`examples/tests/valid/bucket_events.w` compilation and appliance output)

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Monada Contribution License](https://docs.winglang.io/terms-and-policies/contribution-license.html)*.

### CODE REVIEW

**Comment by eladb** [comment]
@tsuf239 PR title should for "fix" should describe the problem being fixed (see [PR tips](https://docs.winglang.io/contributors/pull_requests)).

In this case, I would change it to:

```
fix(sdk): unable to add multiple bucket notifications
```

File: libs/wingsdk/src/target-tf-aws/bucket.ts [hunk 1]
```diff
@@ -51,6 +55,8 @@ export const BUCKET_PREFIX_OPTS: NameOptions = {
 export class Bucket extends cloud.Bucket {
   private readonly bucket: S3Bucket;
   private readonly public: boolean;
+  private notificationTopics: S3BucketNotificationTopic[] = [];
+  private notificationDependencies: ITerraformDependable[] = [];
```
Comment (eladb) [inline libs/wingsdk/src/target-tf-aws/bucket.ts:59]:
```suggestion
  private readonly notificationTopics: S3BucketNotificationTopic[] = [];
  private readonly notificationDependencies: ITerraformDependable[] = [];
```

File: libs/wingsdk/src/target-tf-aws/bucket.ts [hunk 2]
```diff
@@ -82,24 +88,28 @@ export class Bucket extends cloud.Bucket {
...
+
+    this.notificationDependencies.push(handler.permissions);
 
     return handler;
   }
 
+  public _preSynthesize() {
+    super._preSynthesize();
+    if (this.notificationTopics.length) {
+      new S3BucketNotification(this, `S3Object_notifier`, {
```
Comment (eladb) [inline libs/wingsdk/src/target-tf-aws/bucket.ts:105]:
```suggestion
      new S3BucketNotification(this, `S3BucketNotification`, {
```

**Review by eladb — CHANGES_REQUESTED** [review]
Please add a test

**Comment by tsuf239** [comment]
> Please add a test

Relevant tests were edited :)

**Review by Chriscbr — APPROVED** [review]

**Pull request merged**